zrml/prediction-markets/src/lib.rs

```diff
@@ -978,21 +979,38 @@ mod pallet {
         /// and `m` is the number of market ids,
         /// which close at the same time as the specified market.
         #[pallet::weight((
-            T::WeightInfo::reject_market(CacheSize::get(), CacheSize::get()),
+            T::WeightInfo::reject_market(
+                CacheSize::get(),
+                CacheSize::get(),
+                T::MaxRejectReasonLen::get()
```

Comment:
```suggestion
                reject_reason.len() as u32,
```

This ensures that a non-advisor calling this function with a huge vector pays a large fee up front.